storage/buckets/main_test.go

```diff
@@ -15,41 +16,54 @@ import (
 	"golang.org/x/net/context"
 )
 
-var bucketName = fmt.Sprintf("golang-example-buckets-%d", time.Now().Unix())
+var (
+	c          *storage.Client
+	bucketName string
+)
 
-func setup(t *testing.T) *storage.Client {
+func TestMain(m *testing.M) {
 	ctx := context.Background()
 
-	client, err := storage.NewClient(ctx)
+	var err error
+	c, err = storage.NewClient(ctx)
 	if err != nil {
-		t.Fatalf("failed to create client: %v", err)
+		log.Fatalf("failed to create client: %v", err)
 	}
-	return client
+
+	tc, ok := testutil.ContextMain(m)
+	if !ok {
+		log.Println("Skipping tests: GOLANG_SAMPLES_PROJECT_ID must be set")
+		return
+	}
+
+	bucketName = tc.ProjectID + "-storage-tests"
+	// Clean up buckets before running tests.
+	delete(c, bucketName)
+	delete(c, bucketName+"-attrs")
```

Comment:
delete is a built-in function, can we find a better name?

Reply:
Changed to `deleteBucket`. Sample*? Yes.